apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/LiveAppSDKLogic.ts

```diff
@@ -0,0 +1,391 @@
+import { Dispatch } from "redux";
+import { TFunction } from "react-i18next";
+
+import { UserRefusedOnDevice } from "@ledgerhq/errors";
+import {
+  Account,
+  AccountLike,
+  Operation,
+  SignedOperation,
+} from "@ledgerhq/live-common/types/index";
+import {
+  addPendingOperation,
+  getMainAccount,
+  isAccount,
+  isTokenAccount,
+} from "@ledgerhq/live-common/account/index";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { getEnv } from "@ledgerhq/live-common/lib/env";
+import {
+  accountToPlatformAccount,
+  getPlatformTransactionSignFlowInfos,
+} from "@ledgerhq/live-common/platform/converters";
+import {
+  serializePlatformAccount,
+  deserializePlatformTransaction,
+  serializePlatformSignedTransaction,
+  deserializePlatformSignedTransaction,
+} from "@ledgerhq/live-common/platform/serializers";
+import { AppManifest } from "@ledgerhq/live-common/platform/types";
+import {
+  RawPlatformTransaction,
+  RawPlatformSignedTransaction,
+} from "@ledgerhq/live-common/platform/rawTypes";
+import { ToastData } from "@ledgerhq/live-common/notifications/ToastProvider/types";
+
+import { updateAccountWithUpdater } from "../../actions/accounts";
+import { openModal } from "../../actions/modals";
+import { selectAccountAndCurrency } from "../../drawers/DataSelector/logic";
+
+import * as tracking from "./tracking";
+import { MessageData } from "@ledgerhq/live-common/hw/signMessage/types";
+import { prepareMessageToSign } from "@ledgerhq/live-common/hw/signMessage/index";
+
+type WebPlatformContext = {
+  manifest: AppManifest;
+  dispatch: Dispatch;
+  accounts: AccountLike[];
+};
+
+function getParentAccount(account: AccountLike, fromAccounts: AccountLike[]): Account | null {
+  if (isAccount(account)) {
+    return null;
+  }
+
+  const parentAccount = fromAccounts.find(a => a.id === account.parentId);
+  if (!parentAccount) {
+    throw new Error("subaccount has no parent account");
+  }
+  if (!isAccount(parentAccount)) {
+    throw new Error("parentAccount is not of type Account");
+  }
```

Comment:
I notice that we still throw errors here and in some functions of the sdk.
We should choose between throwing errors or rejecting promises, IMO we should use `Promise.reject` instead of `throw` because we already return promises for every functions and it better matches the existing function signatures.
Also if you throw in a `new Promise()` callback the `.catch` won't catch the error.

Reply:
This should not be the responsibility of this function to return a Promise.
I agree we should return a Promise.reject, but at the upper level of the call stack.
Moreover, those error are not functional and in such a case, we have no way to recover from them. We can use some sort of generic technical error for the SDK caller.